http-client-spi/src/main/java/software/amazon/awssdk/http/async/SdkHttpResponseHandler.java

```diff
@@ -49,13 +49,19 @@
     /**
      * Called when an exception occurs during the request/response.
      *
+     * This should be considered a terminal method call. No other method invocations should be expected
+     * on the {@link SdkHttpResponseHandler} after this point.
+     *
      * @param throwable the exception that occurred.
      */
     void exceptionOccurred(Throwable throwable);
 
     /**
      * Called when all parts of the response have been received.
      *
+     * This should be considered a terminal method call. No other method invocations should be expected
```

Comment:
Minor doc suggestion. Remove the should and just state it as a requirement. This is a terminal method call...etc. Also I think there is a public version of this for streaming interfaces, can we double check that makes the expectation clear.

Reply:
Fixed.